### PR TITLE
Iteration 5 wk

### PR DESCRIPTION
**Pull request by whitneykidd**
Updated most_tackles and fewest_tackles methods and their tests.
Both methods now pass spec_harness as well.

### CODE REVIEW

File: lib/season_stats.rb [hunk 1]
```diff
@@ -85,13 +85,22 @@ def calculate_wins(array_games, wins_hash)
   end
 
   def most_tackles(season_param)
-    season_games = season_game_teams(season_param).max_by { |team| team.tackles }
-    find_name(season_games.team_id)
+    # total_tackles = {}
```
Comment (alex-latham) [inline lib/season_stats.rb:88]: Leftover comment

File: lib/team.rb [hunk 2]
```diff
@@ -1,9 +1,11 @@
 class Team
-  attr_reader :team_id, :franchise_id, :teamname
+  attr_reader :team_id, :franchise_id, :teamname, :abbreviations, :link
```
Comment (alex-latham) [inline lib/team.rb:2]: Good catch. Now was the time to add these two

File: lib/team_stats.rb [hunk 3]
```diff
@@ -5,6 +5,16 @@ def initialize(games, teams, game_teams)
...
 
+  def team_info(team_id)
+    {
+      "team_id" => team_id,
+      "franchiseid" => @teams[team_id].franchise_id.to_i,
+      "teamname" => @teams[team_id].teamname,
+      "abbreviation" => @teams[team_id].abbreviations,
+      "link" => @teams[team_id].link
+      }
+  end
```
Comment (alex-latham) [inline lib/team_stats.rb:16]: rspec has the expected result looking like (see team_name and franchise_id):
```
expected = {
      "team_id" => "18",
      "franchise_id" => "34",
      "team_name" => "Minnesota United FC",
      "abbreviation" => "MIN",
      "link" => "/api/v1/teams/18"
    }
```

File: test/stat_tracker_test.rb [hunk 4]
```diff
@@ -158,4 +158,15 @@ def test_it_returns_name_of_team_with_most_tackles
   def test_it_returns_name_of_team_with_fewest_tackles
     assert_equal "Atlanta United", @stat_tracker.fewest_tackles("20132014")
   end
+
+  def test_it_returns_team_info
+    expected = {
```
Comment (alexlath) [inline test/stat_tracker_test.rb]: same as previous

File: test/stat_tracker_test.rb [hunk 5]
```diff
@@ -158,4 +158,15 @@ def test_it_returns_name_of_team_with_most_tackles
...
+                "team_id" => 1,
+                "franchiseid" => 16,
+                "teamname" => "Chicago Fire",
+                "abbreviation" => nil,
+                "link" => "/api/v1/teams/4"
+                }
+    assert_equal expected, @stat_tracker.team_info(1)
+  end
```
Comment (alex-latham) [inline test/stat_tracker_test.rb:171]: Same thing as previous

File: test/team_stats_test.rb [hunk 6]
```diff
@@ -16,6 +16,22 @@ def setup
...
+
+    expected = {
+                "team_id" => 1,
+                "franchiseid" => 16,
+                "teamname" => "Chicago Fire",
+                "abbreviation" => nil,
+                "link" => "/api/v1/teams/4"
+                }
+    assert_equal expected, @team_stats.team_info(1)
+  end
```
Comment (alex-latham) [inline test/team_stats_test.rb:33]: Same as previous, can take out comment

File: lib/season_stats.rb [hunk 7]
```diff
@@ -85,13 +85,22 @@ def calculate_wins(array_games, wins_hash)
...
+    find_name(team_id)
   end
 
   def fewest_tackles(season_param)
-    season_games = season_game_teams(season_param).min_by { |team| team.tackles }
-    find_name(season_games.team_id)
+    total_tackles = season_game_teams(season_param).reduce(Hash.new(0)) do |total_tackles, game_team|
+      total_tackles[game_team.team_id] += game_team.tackles
+      total_tackles
+    end
```
Comment (Maxwell-Baird) [inline lib/season_stats.rb:101]: Good catch on needing to sum the tackles

File: lib/team.rb [hunk 8]
```diff
@@ -1,9 +1,11 @@
...
 
   def initialize(attributes)
     @team_id = attributes[:team_id].to_i
     @franchise_id = attributes[:franchiseid].to_i
     @teamname = attributes[:teamname]
+    @abbreviations = attributes[:abbreviations]
+    @link = attributes[:link]
```
Comment (tylertomlinson) [inline lib/team.rb:9]: Might need to have it formatted for rspec on team_info

**Pull request merged**